### PR TITLE
Add back pinot spi shading

### PR DESCRIPTION
**Pull request by jasperjiaguo**
Add back pinot spi shading which is removed in https://github.com/apache/pinot/pull/12712
Tested building on local branch.

### CODE REVIEW

File: pinot-spi/src/main/java/org/apache/pinot/spi/utils/PinotReflectionUtils.java [hunk 1]
```diff
@@ -48,6 +48,7 @@ public static Set<Class<?>> getClassesThroughReflection(String regexPattern, Cla
     return getClassesThroughReflection(PINOT_PACKAGE_NAME, regexPattern, annotation);
   }
 
+  @SuppressWarnings("deprecation")
```
Comment (Jackie-Jiang) [inline pinot-spi/src/main/java/org/apache/pinot/spi/utils/PinotReflectionUtils.java:51]: Are the changes in this file related?

File: pinot-spi/pom.xml [hunk 2]
```diff
@@ -33,6 +33,7 @@
   <url>https://pinot.apache.org/</url>
   <properties>
     <pinot.root>${basedir}/..</pinot.root>
+    <shade.phase.prop>package</shade.phase.prop>
```
Comment (Jackie-Jiang) [inline pinot-spi/pom.xml:36]: Seems it is not enabled by default. Do you enable it by explicitly setting the flag?

Comment (jasperjiaguo) [reply pinot-spi/pom.xml:36]: Yes. It used to be enabled by the shading profile. Now that we have removed those we should have this set. I followed the change pattern in pom of pinot-common in https://github.com/apache/pinot/pull/12712

File: pinot-spi/src/main/java/org/apache/pinot/spi/utils/PinotReflectionUtils.java (hunk 1 above)
Comment (jasperjiaguo) [reply pinot-spi/src/main/java/org/apache/pinot/spi/utils/PinotReflectionUtils.java:51]: Ahh not really, but felt proabably better to add the suppress warnings for the previous https://github.com/apache/pinot/pull/12958 as you have suggested.

Comment (Jackie-Jiang) [reply pinot-spi/src/main/java/org/apache/pinot/spi/utils/PinotReflectionUtils.java:51]: Oh, I'd suggest suppressing just that one statement by adding `//noinspection deprecation` above it, then the comment and the suppress are next to each other. Can we keep it separate from this PR?

Comment (jasperjiaguo) [reply pinot-spi/src/main/java/org/apache/pinot/spi/utils/PinotReflectionUtils.java]: Sure we can separate this

File: pinot-spi/pom.xml (hunk 2 above)
Comment (jasperjiaguo) [reply pinot-spi/pom.xml]: changed to disabled by default and tested with mvn build, can you take another look @Jackie-Jiang

**Review by Jackie-Jiang — COMMENTED** [review]
Is this the only places where the behavior is changed in #12712?

To keep the exact same behavior and also keep most of the logic in the root pom, we may add a property: `shade.active.by.default`

@xiangfu0 Can you also take a look?

**Comment by jasperjiaguo** [comment]
> Is this the only places where the behavior is changed in #12712?
> 
> To keep the exact same behavior and also keep most of the logic in the root pom, we may add a property: `shade.active.by.default`
> 
> @xiangfu0 Can you also take a look?

Tried using profile at the top level pom to configure this behaviour with smth like 
```
 <profiles>
    <profile>
      <id>build-shaded-jar</id>
      <activation>
         <activeByDefault>${shade.active.by.default}</activeByDefault>
      </activation>
      <build>
        <plugins>
          <plugin>
            <artifactId>maven-shade-plugin</artifactId>
...
```      
but does not work, any suggestions? @Jackie-Jiang 
Even setting ${shade.active.by.default} to true globally, shaded jars doesn't install by default.

**Comment by xiangfu0** [comment]
Can you just add this <shade.phase.prop>package</shade.phase.prop> in the pinot-spi pom.xml file properties section.

**Comment by xiangfu0** [comment]
Please refer to https://github.com/apache/pinot/pull/12979 for the fix

**Comment by Jackie-Jiang** [comment]
Solved with #12979

**Pull request closed without merging**